modules/standard/CTypes.chpl

```diff
@@ -303,10 +337,19 @@ module CTypes {
     return __primitive("cast", t, x);
   }
   pragma "no doc"
+  inline operator c_ptrConst.:(x:c_ptrConst, type t:c_ptrConst) {
```

Comment:
Did we decide to allow this? The answer may be "yes", but I was remembering that it'd have to go through `c_void_ptr` first or something. As long as it is inadvertent, I'm OK. (Not trying to re-discuss anything)

Reply:
We decided to have a safety warning on it in https://github.com/chapel-lang/chapel/issues/21687, which is implemented in https://github.com/chapel-lang/chapel/pull/21934. These PRs interact, so if this one is merged first, the other will need to then rebase off `main` and add the warning for `c_ptrConst`. If the other gets merged first, I will rebase this one and add the warning here.